soda-core/src/soda_core/model/data_source/data_source.py

```diff
@@ -14,8 +15,13 @@ class DataSourceBase(
     validate_by_name=True,  # Allow to use both field names and aliases when populating from dict
 ):
     name: str = Field(..., description="Data source name")
-    type: str = Field(..., description="Data source type")
+    type: Literal["data_source"]
     # Alias connection -> connection_properties to read that from yaml config
     connection_properties: DataSourceConnectionProperties = Field(
         ..., alias="connection", description="Data source connection details"
     )
+
+    @classmethod
+    def get_class_type(cls) -> str:
```

Comment:
the only thing that is a bit hacky - to get the simple "postgres" string for the registering. That string is available in the plugin register thingie but it's not passed along into the init subclass